components/quake_map.py

```diff
@@ -15,17 +19,84 @@
+    time_step -- The time step as seconds. Earthquakes that happened within
+        the time window of this size are shown.
+    slider_value -- The value of the time slider. Controls the temporal
+        position of the time window for which earthquakes are shown.
+    session_id -- ID of the current session
+    """
+
+    return dl.Map(
+        id='quake-map',
+        center=[33.7, -117.3],
+        zoom=8,
+        children=[
+            dl.TileLayer(
+                url=api_url,
+                attribution=attribution),
+            html.Div(id='test-id', children=[
+                get_event_layer(
+                        min_time, max_time, time_step, slider_value, session_id
+                )])
+        ])
+
+
+def get_event_layer(min_time, max_time, time_step, slider_value, session_id):
+    """Return a LayerGroup that contains earthquakes represented as circles.
+
+    The shown earthquake events are selected based on the argument values.
+
+    Keyword arguments:
+    min_time -- A datetime object representing the start of the time frame
+    max_time -- A datetime object representing the end of the time frame
+    time_step -- The time step as seconds. Earthquakes that happened within
+        the time window of this size are shown.
+    slider_value -- The value of the time slider. Controls the temporal
+        position of the time window for which earthquakes are shown.
+    session_id -- ID of the current session
     """
-    Return the map component.
+    start_time = min_time + timedelta(seconds=slider_value*time_step)
+    end_time = start_time + timedelta(seconds=time_step)
+
+    data = earthquake_data.get_earthquake_data(session_id).data
+    datetimes = get_datetimes(data)
+    filtered_quakes = data[(datetimes > start_time) & (datetimes < end_time)]
+
+    quake_circles = [
+        dl.Circle(
+            center=[quake['LATITUDE'], quake['LONGITUDE']],
+            radius=100,
+            color='red',
+            fillOpacity=0.1,
+            weight=2
+        )
+        for _, quake in filtered_quakes.iterrows()]
+
+    return dl.LayerGroup(id='layer-id', children=quake_circles)
+
+
+def get_datetimes(df):
+    """Extract datetimes from a DataFrame.
+
+    Returns a pandas Series object that contains datetimes
+    parsed from the given DataFrame.
+
+    Keyword arguments:
+    df -- A pandas DataFrame that contains the following columns:
+        YEAR, MONTH, DAY, HOUR, MINUTE, SECOND
     """
-    return html.Div([
-        dl.Map(
-            center=[33.7, -117.3],
-            zoom=8,
-            children=[
-                dl.TileLayer(
-                    url=api_url,
-                    attribution=attribution)
-            ])
-    ])
+    return df.apply(
+        lambda x: datetime(
+            int(x['YEAR']),
+            int(x['MONTH']),
+            int(x['DAY']),
+            int(x['HOUR']),
+            int(x['MINUTE']),
+            int(x['SECOND'])
+        ), axis=1)
```

Comment:
This would be better suited in the `EarthquakeData` class, in which case `pandas` is not needed as an import